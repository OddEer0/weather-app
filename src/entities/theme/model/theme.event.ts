import { createEvent } from "effector"

import { IThemeMode } from "../types"

export const setThemeEvent = createEvent<IThemeMode>()

export const toggleThemeEvent = createEvent()
