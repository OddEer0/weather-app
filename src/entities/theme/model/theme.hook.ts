import { useStore } from "effector-react"
import { useEffect } from "react"

import { setThemeEvent, toggleThemeEvent } from "./theme.event"
import { $theme } from "./theme.store"

export const useTheme = () => {
	const theme = useStore($theme)

	useEffect(() => {
		document.body.dataset.theme = theme
	}, [theme])

	return {
		theme,
		toggleHandler: toggleThemeEvent,
		setTheme: setThemeEvent
	}
}
