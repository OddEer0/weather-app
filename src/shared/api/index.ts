export { api } from "./instance"
export * from "./services"
export type {
	ICurrentWeatherResponse,
	ILocationResponse,
	IForecastWeatherResponse,
	IAstroResponse,
	IForecastDayResponse,
	IForecastHourResponse,
	IForecastResponse,
	IMappedForecast,
	IForecastBase
} from "./types"

export * from "./mappers"
