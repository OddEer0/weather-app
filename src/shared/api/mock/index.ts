export { forecastMock } from "./forecast.mock"
