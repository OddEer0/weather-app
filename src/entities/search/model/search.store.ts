import { combine, createEffect, createStore } from "effector"

import { ISearchCity, searchCityService } from "@/shared/api"

export const fetchSearchFx = createEffect((searchValue: string) => {
	return searchCityService.getCity(searchValue)
})

export const $searchCities = createStore<ISearchCity[] | null>(null).on(
	fetchSearchFx.done,
	(_, result) => result.result
)

export const $fetchSearchCity = combine({
	isLoading: fetchSearchFx.pending,
	cities: $searchCities
})
