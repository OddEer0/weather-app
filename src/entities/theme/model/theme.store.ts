import { createPersistStore } from "@/shared/lib"

import { INIT_THEME } from "../lib"
import { IThemeMode } from "../types"

import { setThemeEvent, toggleThemeEvent } from "./theme.event"

export const $theme = createPersistStore<IThemeMode>(INIT_THEME, { name: "theme" })
	.on(setThemeEvent, (_, payload) => payload)
	.on(toggleThemeEvent, state => (state === "dark" ? "light" : "dark"))
