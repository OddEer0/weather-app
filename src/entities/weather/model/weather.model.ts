import { combine, createEffect, createStore } from "effector"

import { ICurrentWeather, IForecastDay, ILocationResponse, forecastWeatherService } from "@/shared/api"

interface ILatLon {
	lat: number
	lon: number
}

export const fetchWeatherFx = createEffect((payload: ILatLon) => {
	return forecastWeatherService.getForecastWeatherByLatLon(payload.lat, payload.lon)
})

const $isLoading = createStore<boolean>(true).on(fetchWeatherFx.done, () => true)

export const $location = createStore<ILocationResponse | null>(null).on(
	fetchWeatherFx.done,
	(_, result) => result.result.location
)

export const $currentWeather = createStore<ICurrentWeather | null>(null).on(
	fetchWeatherFx.done,
	(_, result) => result.result.current
)

export const $forecastWeather = createStore<IForecastDay[] | null>(null).on(
	fetchWeatherFx.done,
	(_, result) => result.result.forecast.forecastday
)

export const $fetchFullWeather = combine({
	isLoading: $isLoading,
	location: $location,
	weather: $currentWeather,
	forecast: $forecastWeather
})

export const $fetchWeather = combine({
	isLoading: $isLoading,
	weather: $currentWeather
})

export const $fetchLocation = combine({
	isLoading: $isLoading,
	location: $location
})

export const $fetchForecast = combine({
	isLoading: $isLoading,
	forecast: $forecastWeather
})
