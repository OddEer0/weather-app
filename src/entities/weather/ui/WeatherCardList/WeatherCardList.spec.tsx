import { render, screen } from "@testing-library/react"
import { fork } from "effector"

import { forecastMapper } from "@/shared/api"
import { forecastMock } from "@/shared/api/mock"

import { $forecastWeather } from "../../model"

import { WeatherCardList } from "./index"

describe("Should render component", () => {
	it("Should render card list with hour weather data", async () => {
		const scope = fork({ values: new Map([[$forecastWeather, forecastMapper(forecastMock.forecast.forecastday)]]) })
		const forecast = scope.getState($forecastWeather)
		const data = forecast ? forecast[0].hour : null

		render(<WeatherCardList isLoading={false} weathers={data} />)
		expect(screen.getByText(/03:00/i)).toBeInTheDocument()
	})

	it("Should render card list with day weather data", async () => {
		const scope = fork({ values: new Map([[$forecastWeather, forecastMapper(forecastMock.forecast.forecastday)]]) })
		const forecast = scope.getState($forecastWeather)
		const data = forecast ? forecast.map(fore => fore.day) : null

		render(<WeatherCardList isLoading={false} weathers={data} />)
		expect(screen.getByText(/Чт, 18 мая/i)).toBeInTheDocument()
	})
})

describe("Screenshot weather card list screenshots", () => {
	it("Screenshot card list with hour weather data", async () => {
		const scope = fork({ values: new Map([[$forecastWeather, forecastMapper(forecastMock.forecast.forecastday)]]) })
		const forecast = scope.getState($forecastWeather)
		const data = forecast ? forecast[0].hour : null

		const { asFragment } = render(<WeatherCardList isLoading={false} weathers={data} />)
		expect(asFragment()).toMatchSnapshot()
	})

	it("Screenshot card list with day weather data", async () => {
		const scope = fork({ values: new Map([[$forecastWeather, forecastMapper(forecastMock.forecast.forecastday)]]) })
		const forecast = scope.getState($forecastWeather)
		const data = forecast ? forecast.map(fore => fore.day) : null

		const { asFragment } = render(<WeatherCardList isLoading={false} weathers={data} />)
		expect(asFragment()).toMatchSnapshot()
	})
})
