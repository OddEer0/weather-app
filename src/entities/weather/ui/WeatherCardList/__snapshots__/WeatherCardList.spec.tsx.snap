// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Screenshot weather card list screenshots > Screenshot card list with day weather data 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root _stack_44342f css-1i8o3yx-MuiStack-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _card_6a843d _card_44342f css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root _content_6a843d css-46bh2p-MuiCardContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption _date_6a843d css-1sn4lm3-MuiTypography-root"
        >
          Чт, 18 мая
        </span>
        <div
          class="_icon_6a843d"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
            />
          </svg>
        </div>
        <div
          class="_footer_6a843d"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            26°c
          </span>
          <span
            class="MuiTypography-root MuiTypography-caption _after_6a843d css-1sn4lm3-MuiTypography-root"
          >
            14°c
          </span>
        </div>
      </div>
    </div>
    <div
      class="_empty_44342f"
    />
    <div
      class="_empty_44342f"
    />
    <div
      class="_empty_44342f"
    />
    <div
      class="_empty_44342f"
    />
    <div
      class="_empty_44342f"
    />
  </div>
</DocumentFragment>
`;

exports[`Screenshot weather card list screenshots > Screenshot card list with hour weather data 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root _stack_44342f css-1i8o3yx-MuiStack-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _card_6a843d _card_44342f css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root _content_6a843d css-46bh2p-MuiCardContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption _date_6a843d css-1sn4lm3-MuiTypography-root"
        >
          00:00
        </span>
        <div
          class="_icon_6a843d"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
            />
          </svg>
        </div>
        <div
          class="_footer_6a843d _footerCenter_6a843d"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            15°c
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _card_6a843d _card_44342f css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root _content_6a843d css-46bh2p-MuiCardContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption _date_6a843d css-1sn4lm3-MuiTypography-root"
        >
          01:00
        </span>
        <div
          class="_icon_6a843d"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
            />
          </svg>
        </div>
        <div
          class="_footer_6a843d _footerCenter_6a843d"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            15°c
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _card_6a843d _card_44342f css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root _content_6a843d css-46bh2p-MuiCardContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption _date_6a843d css-1sn4lm3-MuiTypography-root"
        >
          02:00
        </span>
        <div
          class="_icon_6a843d"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
            />
          </svg>
        </div>
        <div
          class="_footer_6a843d _footerCenter_6a843d"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            14°c
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _card_6a843d _card_44342f css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root _content_6a843d css-46bh2p-MuiCardContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption _date_6a843d css-1sn4lm3-MuiTypography-root"
        >
          03:00
        </span>
        <div
          class="_icon_6a843d"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
            />
          </svg>
        </div>
        <div
          class="_footer_6a843d _footerCenter_6a843d"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            14°c
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _card_6a843d _card_44342f css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root _content_6a843d css-46bh2p-MuiCardContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption _date_6a843d css-1sn4lm3-MuiTypography-root"
        >
          04:00
        </span>
        <div
          class="_icon_6a843d"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
            />
          </svg>
        </div>
        <div
          class="_footer_6a843d _footerCenter_6a843d"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            14°c
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _card_6a843d _card_44342f css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root _content_6a843d css-46bh2p-MuiCardContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption _date_6a843d css-1sn4lm3-MuiTypography-root"
        >
          05:00
        </span>
        <div
          class="_icon_6a843d"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
            />
          </svg>
        </div>
        <div
          class="_footer_6a843d _footerCenter_6a843d"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            15°c
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _card_6a843d _card_44342f css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root _content_6a843d css-46bh2p-MuiCardContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption _date_6a843d css-1sn4lm3-MuiTypography-root"
        >
          06:00
        </span>
        <div
          class="_icon_6a843d"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
            />
          </svg>
        </div>
        <div
          class="_footer_6a843d _footerCenter_6a843d"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            17°c
          </span>
        </div>
      </div>
    </div>
    <div
      class="_empty_44342f"
    />
    <div
      class="_empty_44342f"
    />
    <div
      class="_empty_44342f"
    />
    <div
      class="_empty_44342f"
    />
    <div
      class="_empty_44342f"
    />
  </div>
</DocumentFragment>
`;
